### PR TITLE
Buffs the uplink thermal safe drill

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

The thermal safe drill from uplink now buffs the user if a security team member / clown / captain is seen by the driller / safe. (Might need to change this to hearers, but then windows block it.)
This grants the user a visual effect, 3 brute / burn healing a second, and 25 stamina healing a second.
The buff is lost / not granted if the driller is more than 9 tile from the safe, to block someone planting the drill in the floor safe at arrivals and bum rushing it to the bridge.
The buff lasts regardless of range for 30 seconds after drilling succeeds, to help the driller grab the documents and get out.
Drill now takes 10 seconds to disarm, so a lone officer can not crawl into the vault and disarm it while appearing stunned / crit
Visual effect is given to ghosts for 2 minutes, as well as a follow link when the assault begins.

The drill has a built in instrument, that can play a song when the security assault begins.

## Why It's Good For The Game

The drill sucks. While it is only 1 TC, and automated, it is very easy to disarm and attracts a lot of attention. The safe kit is only 1 tc, and silent, and one can just get a stethoscope from medbay afterwords.
This change makes the drill a more fun option if you are getting attention. It will keep you more alive, though not immortal, and help greatly with stamina while drilling, and for 30 seconds after the drilling finishes.
Music and visual effects make it more interesting, and a fun reference to what the drill is inspired by, after all.

## Images of changes

https://imgur.com/a/RopRCvp

## Testing

Throwing security at the vault to make sure it triggered properly, and that antag clowns would not set off the assault.

## Changelog
:cl:
tweak: Heavily buffs the uplink thermal safe drill. It now detects security assaulting the vault, and buffs the driller during this time.
tweak: Drill now gives the driller a visual effect during assaults.
tweak: Drill now can play music like an instrument when the assault begins.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/safe.dm [hunk 1]
```diff
@@ -322,6 +330,44 @@ GLOBAL_LIST_EMPTY(safes)
...
+			drill.spotted = TRUE
+			security_assualt_in_progress()
+			return
+
+/obj/structure/safe/proc/security_assualt_in_progress()
+	drill.atom_say("Security spotted. Nanites deployed. Give them <b>hell.</b>")
+	driller.apply_status_effect(STATUS_EFFECT_DRILL_PAYBACK, src)
+	drill.song.start_playing(driller)
+	notify_ghosts("Security assault in progress in [get_area(src)]!", enter_link="<a href=?src=[UID()];follow=1>(Click to jump to!)</a>", source = src, action = NOTIFY_FOLLOW)
+	for(var/mob/dead/observer/O in GLOB.player_list)
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/game/objects/structures/safe.dm:356]: Which part?

[the remark this answers is not in the document]
Comment (lewcc) [reply code/game/objects/structures/safe.dm:356]: Sorry, this and the line below it, outputting every ghost to admins

[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/game/objects/structures/safe.dm:356]: Removed!

File: code/datums/uplink_items/uplink_nuclear.dm [hunk 2]
```diff
@@ -394,10 +394,10 @@
 // DEVICES AND TOOLS
 
 /datum/uplink_item/device_tools/diamond_drill
-	name = "Diamond Tipped Thermal Safe Drill"
-	desc = "A diamond tipped thermal drill with magnetic clamps for the purpose of quickly drilling hardened objects. Guaranteed 100% jam proof."
+	name = "Amplifying Diamond Tipped Thermal Safe Drill"
+	desc = "A diamond tipped thermal drill with magnetic clamps for the purpose of quickly drilling hardened objects. Comes with built in security detection and nanite system, to keep you up if security comes a knocking."
```
Comment (S34NW) [inline code/datums/uplink_items/uplink_nuclear.dm:398]:
```suggestion
	desc = "A diamond tipped thermal drill with magnetic clamps for the purpose of quickly drilling hardened objects. Comes with built in security detection and nanite system, to keep you up if security comes a-knocking."
```

**Review by S34NW — APPROVED** [review]

**Pull request merged**